internetarchive/openlibrary
# [WIP] IA-only auth flow

Note: Before accepting, we need to CR and checkin code into olsystem

Todo:
- some naming needs to change.
- test cases required

Test flows:
*Account Creation*
1) Attempt to create an account on OpenLibrary.org w/ the email of an existing Archive.org user (e.g. foo@archive.org)
2) Attempt to create an account on OpenLibrary.org w/ the email of an existing Open Library username
3) Attempt to create an account on OpenLibrary.org w/ with an email that doesn't exist on IA or OL (e.g. foo+test1@archive.org)

*Login*
1) Attempt to login w/ valid OL credentials (the following should all fail):
- try w/ an OL account which hasn't been activated
- try w/ an OL account which is blocked
- try w/ an OL account which is linked
- try w/ an OL account which is not linked
2) Attempt to login w/ an email that doesn't exist on IA or OL (this should fail -- no such account)
3) Attempt to login w/ a valid IA account...
- which isn't verified/activated (should fail)
- which is already linked to an OL account (should succeed)
- which is unlinked & where no OL account exists with matching email (should succeed)
- which is unlinked & shares the same email as an OL account which is also unlinked (should succeed)
- which is unlinked & shares the same email as an OL account linked to another account (this should fail!)
